cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -716,13 +716,16 @@ private void addSystemArtifact(SystemArtifactInfo systemArtifactInfo) throws Exc
    *
    * @param artifactId the artifact to delete
    * @throws IOException if there was some IO error deleting the artifact
+   * @throws ArtifactNotFoundException if the given artifact does not exist
    * @throws UnauthorizedException if the current user is not authorized to delete the artifact. To delete an artifact,
    *                               a user needs {@link Action#ADMIN} permission on the artifact.
    */
   public void deleteArtifact(Id.Artifact artifactId) throws Exception {
     // for deleting artifacts, users need admin privileges on the artifact being deleted.
     Principal principal = authenticationContext.getPrincipal();
     authorizationEnforcer.enforce(artifactId.toEntityId(), principal, Action.ADMIN);
+    // call getArtifact to ensure the artifact exists; if it does not, an ArtifactNotFoundException will be thrown
+    artifactStore.getArtifact(artifactId);
```

Comment:
Shouldn't be done in here. The validation should be done inside the `ArtifactStore.delete` method by having that to throw `ArtifactNotFoundException`.